assignments/assignment7/faulty-oops.md

```diff
@@ -0,0 +1,74 @@
+# Assignment 7: Fault Analysis
+```console
+# echo "hello world" > /dev/faulty 
+Unable to handle kernel NULL pointer dereference at virtual address 0000000000000000
+Mem abort info:
+  ESR = 0x0000000096000045
+  EC = 0x25: DABT (current EL), IL = 32 bits
+  SET = 0, FnV = 0
+  EA = 0, S1PTW = 0
+  FSC = 0x05: level 1 translation fault
+Data abort info:
+  ISV = 0, ISS = 0x00000045
+  CM = 0, WnR = 1
+user pgtable: 4k pages, 39-bit VAs, pgdp=0000000041c01000
+[0000000000000000] pgd=0000000000000000, p4d=0000000000000000, pud=0000000000000000
+Internal error: Oops: 0000000096000045 [#1] SMP
+Modules linked in: hello(O) faulty(O) scull(O)
+CPU: 0 PID: 107 Comm: sh Tainted: G           O       6.1.44 #1
+Hardware name: linux,dummy-virt (DT)
+pstate: 80000005 (Nzcv daif -PAN -UAO -TCO -DIT -SSBS BTYPE=--)
+pc : faulty_write+0x10/0x20 [faulty]
+lr : vfs_write+0xc8/0x390
+sp : ffffffc008d9bd20
+x29: ffffffc008d9bd80 x28: ffffff8001b34f80 x27: 0000000000000000
+x26: 0000000000000000 x25: 0000000000000000 x24: 0000000000000000
+x23: 000000000000000c x22: 000000000000000c x21: ffffffc008d9bdc0
+x20: 000000558eb4a2b0 x19: ffffff8001eceb00 x18: 0000000000000000
+x17: 0000000000000000 x16: 0000000000000000 x15: 0000000000000000
+x14: 0000000000000000 x13: 0000000000000000 x12: 0000000000000000
+x11: 0000000000000000 x10: 0000000000000000 x9 : 0000000000000000
+x8 : 0000000000000000 x7 : 0000000000000000 x6 : 0000000000000000
+x5 : 0000000000000001 x4 : ffffffc000787000 x3 : ffffffc008d9bdc0
+x2 : 000000000000000c x1 : 0000000000000000 x0 : 0000000000000000
+Call trace:
+ faulty_write+0x10/0x20 [faulty]
```

Comment:
@abth5346 this means the fault occurred 0x10 bytes into a function of length 0x20 in `faulty_write`